core/src/transaction.rs

```diff
@@ -63,32 +63,36 @@ impl OutPoint {
 #[derive(Clone, Default, Serialize, Deserialize, PartialEq, Eq, Hash, Debug, OccupiedCapacity)]
 pub struct CellInput {
     pub previous_output: OutPoint,
+    pub valid_since: u64,
     // Depends on whether the operation is Transform or Destroy, this is the proof to transform
     // lock or destroy lock.
     pub args: Vec<Vec<u8>>,
 }
 
 impl CellInput {
-    pub fn new(previous_output: OutPoint, args: Vec<Vec<u8>>) -> Self {
+    pub fn new(previous_output: OutPoint, valid_since: u64, args: Vec<Vec<u8>>) -> Self {
         CellInput {
             previous_output,
+            valid_since,
             args,
         }
     }
 
-    pub fn new_cellbase_input(block_number: BlockNumber) -> Self {
+    pub fn new_cellbase_input(block_number: BlockNumber, valid_since: u64) -> Self {
```

Comment:
I think that `valid_since` of Cellbase should always be 0, no need to pass the parameter here.

Reply:
Fixed